src/BalancedModel.jl

```diff
@@ -0,0 +1,202 @@
+#=
+This is how the struct and the constructor for the model balancer
+    if it were to support only the probabilistic model type:
```

Comment:
This is helpful, thanks.

```suggestion
This is how the struct and the constructor for the model balancer
would look if it were to support only the probabilistic model type:
```